redirect.go

```diff
@@ -2,10 +2,90 @@ package main
 
 import (
 	"fmt"
-	"os"
+	"net/http"
+	"strconv"
+
+	spinhttp "github.com/fermyon/spin/sdk/go/http"
+)
+
+const (
+	// Default value for HTTP status code
+	DefaultStatusCode int = http.StatusFound
+	// Key for loading desired destination
+	destinationKey string = "destination"
+	// Key for loading desired HTTP status code
+	statusCodeKey string = "statuscode"
 )
 
+func init() {
+	r := NewSpinRedirect()
+	spinhttp.Handle(r.handleFunc)
+}
+
 func main() {
-	dest := os.Getenv("DESTINATION")
-	fmt.Printf("status: 302\nlocation: %s\n\n", dest)
+}
+
+// SpinRedirect is a struct that provides a handleFunc
+// for redirecting to a destination URL using configurable HTTP status code.
+type SpinRedirect struct {
+	cfg ConfigReader
+}
+
+// NewSpinRedirect returns a new SpinRedirect
+func NewSpinRedirect() SpinRedirect {
+	return SpinRedirect{
+		cfg: NewDefaultConfigReader(),
+	}
+}
+
+func (s SpinRedirect) handleFunc(w http.ResponseWriter, r *http.Request) {
+	dest := s.getDestination()
+	code := s.getStatusCode(r.Method)
+
+	w.Header().Set("Location", dest)
+	w.WriteHeader(code)
+}
+
+// getDestination returns the destination URL
+// If no destination is found, an empty string is returned.
+func (s SpinRedirect) getDestination() string {
+	return s.cfg.Get(destinationKey)
+}
+
+// getStatusCode returns the HTTP status code
+// If no status code is found, or if the provided value is invalid,
+// DefaultStatusCode is returned.
+func (s SpinRedirect) getStatusCode(method string) int {
+	str := s.cfg.Get(statusCodeKey)
+	code, err := strconv.Atoi(str)
+	if err != nil {
+		return DefaultStatusCode
+	}
+	if !isValidRedirectStatusCode(code, method) {
+		fmt.Printf("Invalid status code provided: %d. Will use %d instead.\n", code, DefaultStatusCode)
```

Comment:
Currently, this would log on every request to the component.   I'm wondering if there is a possibility of catching this and logging sooner, say in tandem with `spin up`?  Perhaps in `NewSpinRedirect()`?

Reply:
@vdice I checked if I can verify and log during `init`. When I call the validation and reporting capabilities from the `init` function, `tinygo build` breaks reliably with `panic: cannot convert pointer value to byte`. 

TBH, I don't fully understand while reading the config value and writing a proper message to `stdout` breaks the build, so I thought about the following option:

I'd love to detect if the user specified verbosity (`RUST_LOG=spin=debug`). If so, we could log it for every request. (See corresponding issue for reference https://github.com/fermyon/spin/issues/1761)

Reply:
> @vdice I checked if I can verify and log during init. When I call the validation and reporting capabilities from the init function, tinygo build breaks reliably with panic: cannot convert pointer value to byte.

Interesting, I wonder why this is.  I may take a look.  However, I suppose we shouldn't block the added functionality on this detail.  We could certainly revisit on a follow-up, perhaps before the next release of this component...  WDYT?

Reply:
@vdice I removed writing to `stdout` for now. Additionally, I've changed the signature of `getDestination` and `getStatusCode` to return `(string, bool)`. The `bool` indicates if the value provided by the user was valid. If https://github.com/fermyon/spin/issues/1761 will be addressed, we can use that as gate for writing to `stdout`.

Reply:
Sounds great, thanks!